velox/exec/RowNumber.cpp

```diff
@@ -378,8 +345,15 @@ void RowNumber::reclaim(
     return;
   }
 
-  if (hashTableSpiller_) {
-    // Already spilled.
+  // TODO: Add a spill level limit for RowNumber spill.
+  if (spillConfig_->exceedJoinSpillLevelLimit(startPartitionBit())) {
```

Comment:
We can add a flag in RowNumber for this and set it in setupHashTableSpiller?
startPartition_ bit which is set to the one in spill config initially and advance in setupHashTableSpiller?

Check if we have exceeded the max spill level in setupHashTableSpiller after we advance the start partition bit, and save in a flag exceededMaxSpillLevelLimit_.

Records this when we records the stats for hash table spiller? Thanks!

Reply:
Keep the current implementation as is and simplify the recording of spill stats in the future per on our discussion.